specification/recoveryservices/Recoveryservices.Management/models.tsp

```diff
@@ -0,0 +1,1685 @@
+   */
+  resource?: string;
+
+  /**
+   * Operations Name itself.
+   */
+  operation?: string;
+
+  /**
+   * Description of the operation having details of what operation is about.
+   */
+  description?: string;
+}
+
+/**
+ * Class to represent shoebox properties in json client discovery.
+ */
+model ClientDiscoveryForProperties {
+  /**
+   * Operation properties.
+   */
+  serviceSpecification?: ClientDiscoveryForServiceSpecification;
+}
+
+/**
+ * Class to represent shoebox service specification in json client discovery.
+ */
+model ClientDiscoveryForServiceSpecification {
+  /**
+   * List of log specifications of this operation.
+   */
+  logSpecifications?: ClientDiscoveryForLogSpecification[];
+}
+
+/**
+ * Class to represent shoebox log specification in json client discovery.
+ */
+model ClientDiscoveryForLogSpecification {
+  /**
+   * Name of the log.
+   */
+  name?: string;
+
+  /**
+   * Localized display name
+   */
+  displayName?: string;
+
+  /**
+   * Blobs created in customer storage account per hour
+   */
+  blobDuration?: string;
+}
+
+/**
+ * Patch Resource information, as returned by the resource provider.
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model PatchVault extends PatchTrackedResource {
+  /**
+   * Properties of the vault.
+   */
+  properties?: VaultProperties;
+
+  /**
+   * Identifies the unique system identifier for each Azure resource.
+   */
+  sku?: Sku;
+
+  /**
+   * Identity for the resource.
+   */
+  identity?: IdentityData;
+}
+
+/**
+ * Tracked resource with location.
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model PatchTrackedResource extends Azure.ResourceManager.Foundations.Resource {
+  /**
+   * Resource location.
+   */
+  location?: string;
+
+  /**
+   * Resource tags.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  tags?: Record<string>;
+
+  /**
+   * Optional ETag.
+   */
+  etag?: string;
+}
+
+/**
+ * Vault extended information.
+ */
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model VaultExtendedInfo {
+  /**
+   * Integrity key.
+   */
+  integrityKey?: string;
+
+  /**
+   * Encryption key.
+   */
+  encryptionKey?: string;
+
+  /**
+   * Encryption key thumbprint.
+   */
+  encryptionKeyThumbprint?: string;
+
+  /**
+   * Algorithm for Vault ExtendedInfo
+   */
+  algorithm?: string;
+}
+
+/**
+ * Operation Resource
+ */
+model OperationResource {
+  /**
+   * End time of the operation
+   */
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  endTime?: utcDateTime;
```

Comment:
>utcDateTime [](http://example.com/codeflow?start=12&length=11)

pretty sure it should be utcDateTime if these are ARM contracts, but why aren't we using OperationResult or OperationStatus or something from common types definitions?

Reply:
OpreationStatus is another case where I'd really like to see conversion tools trying to move us to common-types by default.

Reply:
Conversion tool is to honor original swagger. We cannot do such improvement.